src/components/themed/WalletSyncCircle.js

```diff
@@ -2,94 +2,94 @@
 
 import { type EdgeCurrencyWallet } from 'edge-core-js'
 import * as React from 'react'
+import { ViewStyle } from 'react-native'
 import Animated, { useAnimatedProps, useSharedValue, withTiming } from 'react-native-reanimated'
-import Svg, { Circle } from 'react-native-svg'
+import Svg, { Path } from 'react-native-svg'
 
-import { useEffect, useMemo } from '../../types/reactHooks.js'
+import { useEffect, useRef } from '../../types/reactHooks.js'
 import { useTheme } from '../services/ThemeContext.js'
 
-const AnimatedCircle = Animated.createAnimatedComponent(Circle)
-const BASE_RATIO = 0.05
-const MAX_RATIO = 0.95
+const AnimatedPath = Animated.createAnimatedComponent(Path)
+
 type Props = {
   // The diameter of the inner currency icon:
   size: number,
-  edgeWallet: EdgeCurrencyWallet
+
+  wallet: EdgeCurrencyWallet
 }
 
 /**
  * Renders the sync progress ratio as part of the `CurrencyIcon` component.
  */
 export const WalletSyncCircle = (props: Props) => {
+  const { size, wallet } = props
   const theme = useTheme()
-  const { size = theme.rem(2), edgeWallet } = props
-  // Animation shared state
-  const syncRatio = useSharedValue(edgeWallet.syncRatio < 0.05 ? 0.05 : edgeWallet.syncRatio)
-  const isDone = useSharedValue(false)
-  const stroke = useSharedValue(theme.walletProgressIconFill)
 
   // Subscribe to the sync ratio:
-  useEffect(
-    () =>
-      edgeWallet.watch('syncRatio', (ratio: number) => {
-        // If already done but needs to resync reset the flags and animations
-        if (isDone.value && ratio < BASE_RATIO) {
-          isDone.value = false
-          stroke.value = theme.walletProgressIconFill
-          syncRatio.value = BASE_RATIO
-        }
-        // If the wallet hasn't fully synced show progress animation
-        if (!isDone.value) {
-          if (ratio === 1) {
-            isDone.value = true
-            stroke.value = theme.primaryText
-          }
-          if (ratio < BASE_RATIO) syncRatio.value = withTiming(BASE_RATIO, { duration: 1000 })
-          else if (ratio > MAX_RATIO && ratio < 1) syncRatio.value = withTiming(MAX_RATIO, { duration: 1000 })
-          else syncRatio.value = withTiming(ratio, { duration: 1000 })
+  const done = useRef(wallet.syncRatio > 0.99)
```

Comment:
This is because of floating-point epsilon errors. Suppose you do `(0.1 + 0.2) / 0.3`. The addition is going to give a result like 2.9999999999... because of the way floating-point math works, and then the division may give a result like 0.99999999.... Thus, we don't want to do `>= 1`, because that may never trigger. Instead, we want `> 0.999` or some other threshold we feel comfortable with.

Reply:
Ah, but this error will only occur when adding floating point values. The plugin should be using integer values to represent progress to avoid this type of error: `(1 + 2) / 3`. If a plugin assumes they could add a float value to a running progress ratio, then that is incorrect implementation and should be fixed.